components/calendar/src/japanese.rs

```diff
@@ -296,4 +432,60 @@ impl Japanese {
         }
         .unwrap_or(FALLBACK_ERA)
     }
+
+    /// Returns the range of dates for a given Japanese era code,
+    /// not handling "bce" or "ce"
+    ///
+    /// Returns (era_start, era_end)
+    fn japanese_era_range_for(
+        &self,
+        era: TinyStr16,
+    ) -> Result<(EraStartDate, Option<EraStartDate>), DateTimeError> {
+        // Avoid linear search by trying well known eras
+        if era == tinystr!(16, "reiwa") {
```

Comment:
Nit (nb): I still think this code is premature optimization (and we should just linear search in reverse temporal order)